src/__tests__/render.test.tsx

```diff
@@ -191,11 +211,121 @@ test('debug changing component', () => {
   // eslint-disable-next-line no-console
   const mockCalls = (console.log as any as ConsoleLogMock).mock.calls;
 
-  expect(stripAnsi(mockCalls[4][0])).toMatchSnapshot(
+  expect(stripAnsi(mockCalls[0][0])).toMatchSnapshot(
     'bananaFresh button message should now be "fresh"'
   );
 });
 
+test('debug with only children prop', () => {
+  jest.spyOn(console, 'log').mockImplementation((x) => x);
+
+  const { debug, getByText } = render(<Banana />);
+
+  debug({ mapProps: ({ children }) => ({ children }) });
+
+  // eslint-disable-next-line no-console
+  const mockCalls = (console.log as any as ConsoleLogMock).mock.calls;
+
+  expect(stripAnsi(mockCalls[0][0])).toMatchSnapshot();
+
+  expect(getByText('Change freshness!')).toBeTruthy();
+});
+
+test('debug with only prop whose value is bananaChef', () => {
+  jest.spyOn(console, 'log').mockImplementation((x) => x);
+
+  const { debug, getByText } = render(<Banana />);
+
+  debug({
+    mapProps: (props) => {
+      const filterProps: Record<string, unknown> = {};
+      Object.keys(props).forEach((key) => {
+        if (props[key] === 'bananaChef') {
+          filterProps[key] = props[key];
+        }
+      });
+      return filterProps;
+    },
+  });
+
+  // eslint-disable-next-line no-console
+  const mockCalls = (console.log as any as ConsoleLogMock).mock.calls;
+
+  expect(stripAnsi(mockCalls[0][0])).toMatchSnapshot();
+
+  expect(getByText('Change freshness!')).toBeTruthy();
```

Comment:
Having `getByText` assertion does not seem to contribute to the test.
```suggestion
```

Reply:
Actually what I was trying to do here was make sure that filtering out props wouldn't really delete them but since the children prop is never filtered it doesn't make sense. I'll replace this by a more explicit and relevant test